From 03ffa15fb1008d47529e0226780952b5e15ef9d2 Mon Sep 17 00:00:00 2001
From: Libo Jin <liboj@codeaurora.org>
Date: Tue, 18 Jul 2017 15:10:19 +0800
Subject: [PATCH 21/30] Fix crash issue when user uses multi-window mode on
 WifiCallingSettings

There is possibility that an activity object is null in
PhoneStateListener#onCallStateChanged because it is called by
 different threads. If the activity object is null, NPE happens.
So, null check is necessary.
---
 src/com/android/settings/WifiCallingSettings.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/com/android/settings/WifiCallingSettings.java b/src/com/android/settings/WifiCallingSettings.java
index cb661ed9d5..0d90c251bd 100644
--- a/src/com/android/settings/WifiCallingSettings.java
+++ b/src/com/android/settings/WifiCallingSettings.java
@@ -87,6 +87,9 @@ public class WifiCallingSettings extends SettingsPreferenceFragment
         @Override
         public void onCallStateChanged(int state, String incomingNumber) {
             final SettingsActivity activity = (SettingsActivity) getActivity();
+            if (activity == null) {
+                return;
+            }
             boolean isNonTtyOrTtyOnVolteEnabled = ImsManager
                     .isNonTtyOrTtyOnVolteEnabled(activity);
             final SwitchBar switchBar = activity.getSwitchBar();
-- 
2.11.0

